src/test/java/testsuite/integration/hikariCP/HikariCPSQLException.java

```diff
@@ -0,0 +1,18 @@
+package testsuite.integration.hikariCP;
+
+import com.zaxxer.hikari.SQLExceptionOverride;
+
+import java.sql.SQLException;
+
+public class HikariCPSQLException implements SQLExceptionOverride {
+    public Override adjudicate(final SQLException sqlException) {
+        String sqlState = sqlException.getSQLState();
+        if (sqlState.equalsIgnoreCase("08S02") ||
+            sqlState.equalsIgnoreCase("08007") ||
+            sqlState.equalsIgnoreCase("08001")) {
```

Comment:
I'm not sure about this error code. This errors means that failover has failed and the current connection is closed. I don't see any point to keep such connection in connection pool.